ui/components/Dashboard/ConnectionCharts.js

```diff
@@ -1,24 +1,25 @@
-import React , { useEffect,useState } from "react";
-import {  Box , Typography } from "@material-ui/core";
+import React , { useEffect, useState } from "react";
+import { CircularProgress, Box , Typography } from "@material-ui/core";
 import { donut } from "billboard.js";
 import BBChart from "../BBChart";
-import { dataToColors , isValidColumnName } from "../../utils/charts"
+import { dataToColors , isValidColumnName } from "../../utils/charts";
 import { getConnectionStatusSummary } from "../../api/connections";
 import ConnectClustersBtn from "../General/ConnectClustersBtn";
 import Link from "next/link";
 
-
 export default function ConnectionStatsChart({ classes }) {
-
-  const [chartData,setChartData] = useState([])
+  const [chartData, setChartData] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);  // New isLoading state
 
   useEffect(() => {
+    setIsLoading(true);  // Set loading state to true when fetch begins
```

Comment:
It's not required if you initialize the state with `true`. Also handle an error case as well for better UX